### PR TITLE
Fix broken install key check

### PR DESCRIPTION
**Pull request by ajvb**
Currently, we place the datadog install key every puppet run:

```
Notice: /Stage[main]/Datadog_agent::Ubuntu/Datadog_agent::Ubuntu::Install_key[935F5A436A5A6E8788F0765B226AE980C7A7DA52]/Exec[key 935F5A436A5A6E8788F0765B226AE980C7A7DA52]/returns: executed successfully
Notice: /Stage[main]/Datadog_agent::Ubuntu/Datadog_agent::Ubuntu::Install_key[A2923DFF56EDA6E76E55E492D3A80E30382E94DE]/Exec[key A2923DFF56EDA6E76E55E492D3A80E30382E94DE]/returns: executed successfully
```

This fixes that.

### CODE REVIEW

**Comment by ajvb** [comment]
From Jeff:
```
a better PR might be to have them use `puppetlabs-apt` instead
of rolling their own apt repo setup
```

File: lib/puppet/parser/functions/fingerprint_formatter.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      raise Puppet::ParseError, 'fingerprint_formatter requires a fingerprint to format as an argument.'
+    end
+
+    fingerprint_broken_up = []
+    beginning, ending = 0, 3
+    10.times do
+      fingerprint_broken_up << fingerprint.split('')[beginning..ending]
+      beginning += 4
+      ending += 4
+    end
```
Comment (ffleming) [inline lib/puppet/parser/functions/fingerprint_formatter.rb:17]: Maybe I'm missing something but:
```
print = '6F6B15509CF8E59E6E469F327F438280EF8D349F'
> "6F6B15509CF8E59E6E469F327F438280EF8D349F"
print.split(//).each_slice(4).to_a.map(&:join).join(' ')
=> "6F6B 1550 9CF8 E59E 6E46 9F32 7F43 8280 EF8D 349F"
```

Comment (ffleming) [reply lib/puppet/parser/functions/fingerprint_formatter.rb:17]: (there might be a prettier way to do that on a string without breaking it into a an array with single-char elements tho)

Comment (ajvb) [reply lib/puppet/parser/functions/fingerprint_formatter.rb:17]: ❤️  thank you. Not missing anything, I just can't ruby.

**Pull request closed without merging**